mm2src/database/stats_nodes.rs

```diff
@@ -0,0 +1,82 @@
+/// This module contains code to work with nodes table for stats collection in MM2 SQLite DB
+use crate::mm2::lp_stats::{NodeInfo, NodeVersionStat};
+use common::log::debug;
+use common::mm_ctx::MmArc;
+use common::rusqlite::{Error as SqlError, Result as SqlResult, NO_PARAMS};
+use std::collections::hash_map::HashMap;
+
+pub const CREATE_NODES_TABLE: &str = "CREATE TABLE IF NOT EXISTS nodes (
+    id INTEGER NOT NULL PRIMARY KEY,
+    name VARCHAR(255) NOT NULL UNIQUE,
+    address VARCHAR(255) NOT NULL,
+    peer_id VARCHAR(255) NOT NULL UNIQUE
+);";
+
+pub const CREATE_STATS_NODES_TABLE: &str = "CREATE TABLE IF NOT EXISTS stats_nodes (
+    id INTEGER NOT NULL PRIMARY KEY,
+    name VARCHAR(255) NOT NULL,
+    version VARCHAR(255) NOT NULL,
+    timestamp INTEGER NOT NULL
+);";
+
+const INSERT_NODE: &str = "INSERT INTO nodes (name, address, peer_id) VALUES (?1, ?2, ?3)";
+
+const DELETE_NODE: &str = "DELETE FROM nodes WHERE name = ?1";
+
+const SELECT_PEERS_ADDRESSES: &str = "SELECT peer_id, address FROM nodes";
+
+const SELECT_PEERS_NAMES: &str = "SELECT peer_id, name FROM nodes";
+
+const INSERT_STAT: &str = "INSERT INTO stats_nodes (name, version, timestamp) VALUES (?1, ?2, ?3)";
+
+pub fn insert_node_info(ctx: &MmArc, node_info: &NodeInfo) -> SqlResult<()> {
+    debug!("Inserting info about node {} to the SQLite database", node_info.name);
+    let params = vec![
+        node_info.name.clone(),
+        node_info.address.clone(),
+        node_info.peer_id.clone(),
+    ];
+    let conn = ctx.sqlite_connection();
+    conn.execute(INSERT_NODE, &params).map(|_| ())
+}
+
+pub fn delete_node_info(ctx: &MmArc, name: String) -> SqlResult<()> {
+    debug!("Deleting info about node {} from the SQLite database", name);
+    let params = vec![name];
+    let conn = ctx.sqlite_connection();
+    conn.execute(DELETE_NODE, &params).map(|_| ())
+}
+
+pub fn select_peers_addresses(ctx: &MmArc) -> SqlResult<Vec<(String, String)>, SqlError> {
+    let conn = ctx.sqlite_connection();
+    let mut stmt = conn.prepare(SELECT_PEERS_ADDRESSES)?;
+    let peers_addresses = stmt
+        .query_map(NO_PARAMS, |row| Ok((row.get(0)?, row.get(1)?)))?
+        .collect::<SqlResult<Vec<(String, String)>>>()?;
+
+    Ok(peers_addresses)
+}
+
+pub fn select_peers_names(ctx: &MmArc) -> SqlResult<HashMap<String, String>, SqlError> {
+    let conn = ctx.sqlite_connection();
+    let mut stmt = conn.prepare(SELECT_PEERS_NAMES)?;
+    let peers_names = stmt
+        .query_map(NO_PARAMS, |row| Ok((row.get(0)?, row.get(1)?)))?
+        .collect::<SqlResult<HashMap<String, String>>>();
+
+    peers_names
+}
+
+pub fn insert_node_version_stat(ctx: &MmArc, node_version_stat: &NodeVersionStat) -> SqlResult<()> {
```

Comment:
According to the current use case, I think we can use `node_version_stat: NodeVersionStat` to avoid cloning in this fn. In the future, the caller might explicitly clone the entire struct if required.